code/datums/jobs/squads.dm

```diff
@@ -389,3 +403,54 @@
 		if(squad.assign_initial(player, job, latejoin))
 			return TRUE
 	return FALSE
+
+GLOBAL_LIST_EMPTY_TYPED(custom_squad_radio_freqs, /datum/squad)
+///initializes a new custom squad. all args mandatory
+/proc/create_squad(squad_name, squad_color, mob/living/carbon/human/creator)
+	//Create the squad
+	if(!squad_name)
+		return
+	if(!squad_color)
+		return
+	if(!ishuman(creator))
+		return
+	var/freq = FREQ_CUSTOM_SQUAD_MIN + 2 * length(GLOB.custom_squad_radio_freqs)
+	if(freq > FREQ_CUSTOM_SQUAD_MAX)
+		return
+
+	var/new_id = lowertext(squad_name) + "_squad"
+	if(SSjob.squads[new_id])
+		return
+
+	var/squad_faction = creator.faction
+	var/datum/squad/new_squad = new(squad_color)
+	new_squad.name = squad_name
```

Comment:
the tracking id stuff is set in New and requires the squad name be already set.
I know you manually call this proc later, but init_squad will still init a squad which is probably not ideal. Not the end of the world tho.

Reply:
made it pass to new instead like color